cmd/adapter/adapter.go

```diff
@@ -334,12 +356,35 @@ func main() {
 		klog.Fatalf("unable to install resource metrics API: %v", err)
 	}
 
+	config, err := cmd.Config()
+	if err != nil {
+		klog.Fatalf("unable to get configuration of custom metrics apiserver: %v", err)
+	}
+
+	apiServerConfig(config)
+	cmd.setupMetricsServer()
+
 	// run the server
 	if err := cmd.Run(stopCh); err != nil {
 		klog.Fatalf("unable to run custom metrics adapter: %v", err)
 	}
 }
 
+func apiServerConfig(config *customexternalmetrics.Config) {
+	config.GenericConfig.EnableMetrics = false
+}
+
+func (cmd PrometheusAdapter) setupMetricsServer() {
```

Comment:
it would be better to take the PrometheusAdapter by pointer here

Reply:
Done